### ISSUE TITLE
[packagers] NPE when trying to use homebrew cask

### ISSUE THREAD

**Issue by crschnick** [jreleaser/jreleaser#1321]
I'm trying to configure homebrew with the cask feature, but I always get an NPE. Here's a configuration:
  ```
  def proj = project
  
  jreleaser {
      packagers {
          brew {
              active = 'ALWAYS'
              continueOnError = false
              templateDirectory = 'homebrew'
              repoTap {
                  active = 'ALWAYS'
                  repoOwner = 'xpipe-io'
                  name = 'homebrew-tap'
                  tagName = '{{distributionName}} {{tagName}}'
                  username = 'crschnick'
              }
              formulaName = 'xpipe'
              multiPlatform = false
  
              // Cask configuration.
              // Requires artifact with extension `.dmg` or `.pkg`.
              // Defaults are shown.
              //
              cask {
                  enabled = true
                  name = 'xpipe'
                  displayName = 'X-Pipe'
                  uninstall = [
                          pkgutil: [
                                  'io.xpipe.xpipe'
                          ]
                  ]
              }
          }
      }
  }
```

[trace.log](https://github.com/jreleaser/jreleaser/files/11224933/trace.log)

**Comment by aalmiray** [comment]
The stacktrace is 

```
[DEBUG]   [validation] packagers
[DEBUG]   [validation] distributions
[DEBUG]   [validation] distribution.installer
[DEBUG]   [validation] distribution.installer.appImage
[DEBUG]   [validation] - disabled
[DEBUG]   [validation] distribution.installer.asdf
[DEBUG]   [validation] - disabled
[DEBUG]   [validation] distribution.installer.brew
[DEBUG]   [validation] distribution.installer.brew.cask
java.lang.NullPointerException: Cannot invoke "String.toLowerCase(java.util.Locale)" because the return value of "org.jreleaser.model.internal.packagers.BrewPackager.getResolvedFormulaName(org.jreleaser.model.internal.JReleaserContext)" is null
	at org.jreleaser.model.internal.validation.packagers.BrewPackagerValidator.validateCask(BrewPackagerValidator.java:218)
	at org.jreleaser.model.internal.validation.packagers.BrewPackagerValidator.validateBrew(BrewPackagerValidator.java:92)
	at org.jreleaser.model.internal.validation.distributions.DistributionsValidator.validateDistribution(DistributionsValidator.java:210)
	at org.jreleaser.model.internal.validation.distributions.DistributionsValidator.validateDistributions(DistributionsValidator.java:87)
	at org.jreleaser.model.internal.JReleaserModelValidator.validateModel(JReleaserModelValidator.java:79)
	at org.jreleaser.model.internal.JReleaserModelValidator.validate(JReleaserModelValidator.java:57)
	at org.jreleaser.model.internal.JReleaserContext.validateModel(JReleaserContext.java:480)
	at org.jreleaser.engine.context.ModelValidator.validate(ModelValidator.java:47)
	at org.jreleaser.workflow.WorkflowImpl.<init>(WorkflowImpl.java:45)
	at org.jreleaser.workflow.Workflows.packageRelease(Workflows.java:120)
	at org.jreleaser.gradle.plugin.tasks.JReleaserPackageTask.performAction(JReleaserPackageTask.groovy:43)
```

Resolved model

```
packagers:
    brew:
        enabled: true
        active: ALWAYS
        continueOnError: false
        templateDirectory: /Users/crschnick/dev/xpipe/xpipex/dist/homebrew
        commitAuthor:
            name: jreleaserbot
            email: jreleaser@kordamp.org
        formulaName: xpipe
        multiPlatform: false
        tap:
            enabled: true
            active: ALWAYS
            owner: xpipe-io
            name: homebrew-tap
            tagName: {{distributionName}} {{tagName}}
            branch: HEAD
            branchPush: HEAD
            username: crschnick
            token: ************
        cask:
            enabled: true
            name: xpipe
            displayName: X-Pipe
            uninstall:
                pkgutil:
                    io.xpipe.xpipe

distributions:
    installer:
        enabled: true
        active: ALWAYS
        type: NATIVE_PACKAGE
        executable:
            name: installer
            windowsExtension: bat
        platform:
        artifacts:
            artifact 0:
                path: /Users/crschnick/dev/xpipe/xpipex/dist/build/dist/artifacts/xpipe-installer-linux-x86_64.rpm
                platform: linux-x86_64
                extraProperties:
                    optional: true
            artifact 1:
                path: /Users/crschnick/dev/xpipe/xpipex/dist/build/dist/artifacts/xpipe-installer-macos-x86_64.pkg
                platform: osx-x86_64
                extraProperties:
                    optional: true
        brew:
            enabled: true
            active: ALWAYS
            continueOnError: false
            commitAuthor:
            multiPlatform: false
            tap:
                enabled: false
                name: homebrew-tap
                token: **unset**
            cask:
                enabled: true
                pkgName: xpipe-installer-macos-x86_64.pkg
```

**Comment by crschnick** [comment]
Is there any way to use this fix in gradle before the next release? Like are there snapshot releases for the plugin?

**Comment by aalmiray** [comment]
We don't publish snapshot releases for plugins. However, the next release is just around the corner, by the end of April 2023.

**Comment by crschnick** [comment]
Is there an easy way to maybe run the jreleaser CLI early access version using the model defined in gradle? In other words, is for example the model that is shown in the trace.log somehow exportable to be used outside of gradle?

Because I I think there's some chance for another issue to occur with homebrew as I didn't get past the configuration validation stage.

**Comment by aalmiray** [comment]
Not at the moment. The printed model looks a lot like YAML but it's not compatible with the actual YAML format ingested by the CLI.

**Comment by aalmiray** [comment]
Idea: add a `jreleaserExport` task to Gradle and `jreleaser:export` to Maven. This would export the _resolved_ model (into a target format [YAML, JSON, TOML].

Bear in mind that the resolved model will contain much more information than the original DSL, similar to what you see when invoking `jreleaserConfig`.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.6.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.6.0))
